tofhir-server/src/main/scala/io/tofhir/server/service/JobService.scala

```diff
@@ -77,6 +77,19 @@ class JobService(jobRepository: IJobRepository) extends LazyLogging {
     jobRepository.runJob(projectId, jobId,mappingUrls)
   }
 
+  /**
+   * Tests the given mapping task by running it with mapping job configurations (i.e. source data configurations) and
+   * returns its results
+   *
+   * @param projectId project id the job belongs to
+   * @param jobId job id
+   * @param mappingTaskTest mapping task test to be executed
+   * @return
+   */
+  def testMappingWithJob(projectId: String, jobId: String, mappingTaskTest: FhirMappingTaskTest): Future[Future[Seq[FhirMappingResult]]] = {
```

Comment:
In parallel to my comment above, I would extract the mapping task out and provide it as a standalone parameter